### ISSUE TITLE
Where do you use the class weights for energy calculation during inference?

### ISSUE THREAD

**Issue by KostadinovShalon** [deeplearning-wisc/vos#53]
Hi,

I've been looking around in the code and for the inference predictor, it seems that you only use directly the class logits for getting the ood_score, specifically, in [here](https://github.com/deeplearning-wisc/vos/blob/a449b03c7d6e120087007f506d949569c845b2ec/voc_coco_plot.py#L48C26-L48C100):

`id_score = -args.T * torch.logsumexp(torch.stack(id_data['inter_feat'])[:, :-1] / args.T, dim=1).cpu().data.numpy()`

you use the torch.logsumexp, which does not use the class weights for the energy in Eq. 6 (as in your own implementation of log_sum_exp in roihead_gmm). I was wondering why is that? Is it really not necessary?

Thanks!

**Comment by d12306** [comment]
hi, @KostadinovShalon , thanks for your question! During test time, the data that we want to calculate the score on is different so it is not meaningful if we use the same class weights as training.  we have also tested using the class weights for inference, the result is getting worse, so the original score should be fine.

**Comment by KostadinovShalon** [comment]
Thanks for the reply. Yeah I see, but within the same context, I see you don't use the "phi" function in the inference either. I mean, the logistic_regression layer is not applied, it is just the energy that is being used for getting the metrics. I might be a bit lost, but shouldn't you use this layer in inference?
